hugo/content/guides/scoping/class-member.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "Class Member Scoping"
+weight: 200
+---
+
+In this guide we will take at member based scoping. It's a mechanism you are likely familiar with from object oriented languages such as Java, C# and JavaScript:
+
+```java
+class A {
+    public B b;
+}
+
+class B {
+    public String value;
+}
+
+void test() {
+    A a = new A();
+    B b = a.b; // Refers to the `b` defined in class `A`
+    String value = b.value; // Refers to the `value` defined in class `B`
+}
+```
+
+Member based scoping like this requires not only a modification of the default scoping provider, but also some other prerequisites.
+This includes a member call mechanism in your grammar and a rudamentary type system.
+For this guide, we will use excerpts from the [langium-lox](https://github.com/langium/langium-lox) project.
+It implements a strongly-typed version of the [Lox language](https://craftinginterpreters.com/the-lox-language.html) from the popular book [Crafting Interpreters](https://craftinginterpreters.com/).
+
+We first start with the `MemberCall` grammar rule. It references one of our `NamedElements`, which could be variable declarations, functions, classes or methods and fields of those classes. Additionally, we allow function calls. Note that the grammar has no notion of whether these elements can actually be executed as functions. Instead, we always allow function calls on every named element and simply provide validation errors in case they were used erroneously. After parsing the first member call, we continue parsing further members as long as the input text provides us with further references to elements, separated by dots.
+
+```ts
+type NamedElement = FunctionDeclaration | VariableDeclaration | MethodMember | FieldMember | Class;
+
+MemberCall:
```

Comment:
This is a pretty dense parser rule. It might make sense to split it into 3 parts, introducing how each element of the rule works at a time, and then indicating that these are all part of the same `MemberCall` rule.